src/hooks/useGetAllReleases.ts

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { PipelineRunLabel } from '../consts/pipelinerun';
+import { K8sQueryListResourceItems } from '../k8s';
+import { ReleaseModel } from '../models';
+import { ReleaseKind } from '../types';
+
+export interface Release {
+  releasePlan: string;
+  application: string;
+  tenant: string;
+  name: string;
+  completionTime: string;
+  status: string;
+  component: string;
+}
+
+/**
+ * Get all the valid releases across the namespaces that you can see.
+ *
+ * @returns the array of the Release interface.
+ */
+export const useGetAllReleases = (loaded, namespaces) => {
+  const [releases, setData] = React.useState<Release[]>([]);
+
+  React.useEffect(() => {
+    const fetchAll = async (): Promise<Release[]> => {
+      const results: Release[] = [];
+      const namespacePromises = namespaces.map(async (namespace) => {
+        const nsReleases = await K8sQueryListResourceItems<ReleaseKind[]>({
```

Comment:
fetching from all the applications in a namespace can slow down the UI a lot. We should have pagination or infinite scrolling implemented here?

Reply:
If it's should not be part of this PR, Then put this page under a feature flag. That would indicate that it's unstable.

Reply:
@xiangge did you check if using a "feature flag" would make sense for the "release monitor" feature? if so, then you could take a look at our [feature flag docs](https://github.com/konflux-ci/konflux-ui/blob/main/docs/feature-flags.md).

Reply:
Make sense, I did that. Please check in the latest PR.